packages/address-book-controller/src/AddressBookController.ts

```diff
@@ -153,6 +195,174 @@ export class AddressBookController extends BaseController<
 
     return true;
   }
+
+  /**
+   * Initialization step which optionally listens to changes to frequent RPC
+   * chain IDs in {@link PreferencesController}) state and ensures that
+   * addresses are filed under the correct chain.
+   */
+  #handleChainIdChanges() {
+    const { getProvider } = this.config;
+
+    if (
+      this.config.syncWithRpcChanges &&
+      this.config.onPreferencesStateChange !== undefined &&
+      getProvider !== undefined
+    ) {
+      this.config.onPreferencesStateChange(
+        (newPreferencesState, previousPreferencesState) => {
+          return this.#handlePreferencesStateChange(
+            previousPreferencesState,
+            newPreferencesState,
+            getProvider,
+          ).catch((error) => {
+            // TODO: Should this use log?
```

Comment:
[Previously we used the `loglevel` package to log messages](https://github.com/MetaMask/metamask-extension/blob/7e7be2769df7c24413dc6ac92404a31ad2d112f2/app/scripts/controllers/preferences.js). This is a new message that we didn't print before, so technically, we can do whatever we want. Considering we want to know whether this promise fails given that the error will be swallowed otherwise, I figured it was better to use `console.error`, but I could be convinced otherwise.